plugins/auth-backend-module-guest-provider/src/resolvers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { SignInResolver } from '@backstage/plugin-auth-node';
+
+/**
+ * Provide a default implementation of the user to resolve to. By default, this
+ *  is `user:default/guest`. We will attempt to get that user if they're in the
+ *  catalog. If that user doesn't exist in the catalog, we will still create a
+ *  token for them so they can keep viewing.
+ */
+export const signInAsGuestUser: SignInResolver<{}> = async (_, ctx) => {
+  const userRef = stringifyEntityRef({
+    kind: 'user',
+    name: 'guest',
```

Comment:
I like `user:development/guest`, I think there's a longer RFC for updating that globally, but this could be a good module to enforce that default and start moving folks towards it.

The auth context is a little weird here, either we allow configuring this at the `auth.*` level or the `auth.providers.guest.*` level. I'm open to either, but if we get to a future where we allow this to be easily configurable for the guest module, would we want this to be configurable across the application? Are there any benefits for that, ie from an auth perspective?